### PR TITLE
Change hook name to proper model name.

### PR DESCRIPTION
**Pull request by ryanbrinn**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2052

# What does this Pull Request do?
Currently the islandora_compound_object_compoundCModel_islandora_derivative hook  https://github.com/Islandora/islandora_solution_pack_compound/blob/7.x/islandora_compound_object.module#L273 is named wrong, the Cmodel should be islandora_compoundCModel NOT compoundCModel. This renaming allows the hook to fire properly.

# What's new?
Renamed islandora_compound_object_compoundCModel_islandora_derivative hook to islandora_compound_object_islandora_compoundCModel_islandora_derivative.

Example:
N/A

# How should this be tested?
This appears to be an edge case since there is redundancy built in for most use-cases. i.e. https://github.com/Islandora/islandora_solution_pack_compound/blob/7.x/islandora_compound_object.module#L292 was masking this bug.   
To reproduce the problem, ingest a compound object with some debugging around islandora_compound_object_compoundCModel_islandora_derivative. The derivative should not fire. 
Once the fix is employed, the derivative should fire.

# Additional Notes:
N/A

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by DiegoPino** [comment]
@ryanbrinn can you please edit the default template and add the description of what this pull does? thanks!

**Comment by ryanbrinn** [comment]
To clarify, the redundancy between compound deriving from it's children and children propagating to their compound kicks in in all normal use cases. This bug occurred in a very specific case where children were being associated to compounds before the compounds were ingested. Thus, derivatives were firing in an order unique to this specific case.

**Comment by DiegoPino** [comment]
@ryanbrinn this is very interesting. Will test during the week. Thanks!!

**Comment by DiegoPino** [comment]
@ryanbrinn this is very interesting. Will test during the week. Thanks!!

**Review by willtp87 — APPROVED** [review]

**Comment by nhart** [comment]
Hi @DiegoPino, have you had a chance to test this yet?

File: islandora_compound_object.module
Comment (DiegoPino) [inline islandora_compound_object.module]: This should say: "* Implements hook_CMODEL_islandora_derivative()".

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Minor function phpdoc issue

File: islandora_compound_object.module
Comment (ryanbrinn) [reply islandora_compound_object.module]: Done.

**Comment by nhart** [comment]
good catch @DiegoPino.  Ryan has committed the fix. Cheers, Nelson

**Comment by DiegoPino** [comment]
@ryanbrinn since that hook was not firing, is there any chance of race conditions of both derivatives hooks firing?Does the ingest order triggers the difference? I mean, it is not about this pull (this one is right) but about your parent previous one, but since this is the one that makes the call actually available I'm asking here.

**Comment by ryanbrinn** [comment]
It's two different ways of doing the same thing. One says; look at my first child and copy it's thumbnail, the other says; if I'm the first child, copy my thumbnail to the parent.

**Comment by nhart** [comment]
Any further comments @DiegoPino? Any other @Islandora/7-x-1-x-committers care to chime in?

**Comment by DiegoPino** [comment]
@ryanbrinn right i'm now udpating our Travis-CI YAML on all repositories, so let me fix the rest and I will merge because your pull actually fixes the less server resources demanding hook.

@nhart my  concern are about both hooks firing, which actually happens. But this one is the one that does not convinces me:
https://github.com/Islandora/islandora_solution_pack_compound/blob/3645b627647410e6ddc9326e67886c4638f7768b/includes/derivatives.inc#L84
It is kinda annoying since it basically fires for every existing object if this TN derivative generation is enabled. And its an quite expensive function. I mean if i want TN via derivatives (the normal hook, the cmodel one, the one that this pull fixes) then i have to enable the switch.
**Same switch** also enables the "check parent and add a thumbnail", which means if i'm ingesting a book (for example) with many many pages that thing will kick in lowering my performance quite a lot. Actually i don't like it.

**Review by DiegoPino — APPROVED** [review]

**Comment by nhart** [comment]
Any @Islandora/7-x-1-x-committers care to hit the merge button?

**Comment by jonathangreen** [comment]
@nhart @ryanbrinn can you rebase this on top of the latest 7.x? Travis is failing because of the PHP 5.3 issue fixed in 7.x. I'd like to see Travis be green before I hit merge.

**Comment by ryanbrinn** [comment]
@jonathangreen I merged the latest 7.x changes into this branch. looks like travis builds are passing. we should be gtg.

**Comment by nhart** [comment]
Thank you @jonathangreen!

**Pull request merged**
